lib/rmt/downloader.rb

```diff
@@ -204,11 +205,24 @@ def handle_checksum_verification!(checksum_type, checksum_value, download_path)
     end
   end
 
-  def http_error(remote_file, http_code)
-    RMT::Downloader::Exception.new(
-      _('%{file} - HTTP request failed with code %{code}') % { file: remote_file, code: http_code },
-      http_code
-    )
+  def invalid_response?(response)
+    response.code != 200 || (response.return_code && response.return_code != :ok)
```

Comment:
This might cause problems when the it's a success code different from 200. For example, `DELETE` requests return 204 as success. You should set the success code to a range 2xx-299. See more [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status)

Reply:
`invalid_response?` might return true even when the request is successful, for example `DELETE`-ing (which will return 204). This is not correct and will cause bugs in the future.